File: RayTracing.m
clear;close all;clc;figure, hold on
%% Setup
Way='L';
[Xobj,Yobj]=obj(15,1);

[Ximg,Yimg] = lens(Xobj,Yobj,30,-10,Way,1);
imagePlane(Ximg,Yimg);

[Ximg,Yimg,Way] = mirror(Ximg,Yimg,60,10,Way);
imagePlane(Ximg,Yimg);

[Ximg,Yimg] = lens(Ximg,Yimg,30,-10,Way,0);
imagePlane(Ximg,Yimg);
m=Yimg/Yobj;

%%
lim=xlim;
plot([lim(1) lim(2)],[0 0],'k','LineWidth', 1.5)
set(gcf,'renderer','painters');

File: ray_tracing.py
import matplotlib.pyplot as plt

from obj import obj
from lens import lens
from mirror import mirror
from image_plane import image_plane


def main():
    plt.figure()

    way = 'L'
    Xobj, Yobj = obj(15, 1)

    Ximg, Yimg = lens(Xobj, Yobj, 30, -10, way, 1)
    image_plane(Ximg, Yimg)

    Ximg, Yimg, way = mirror(Ximg, Yimg, 60, 10, way)
    image_plane(Ximg, Yimg)

    Ximg, Yimg = lens(Ximg, Yimg, 30, -10, way, 0)
    image_plane(Ximg, Yimg)
    m = Yimg / Yobj

    lim = plt.xlim()
    plt.plot([lim[0], lim[1]], [0, 0], 'k', linewidth=1.5)
    plt.gcf().canvas.draw()  # ensures limits are respected before showing
    plt.show()


if __name__ == "__main__":
    main()
